src/uri.rs

```diff
@@ -30,7 +30,7 @@ use Error;
 /// scheme          authority                 path            query         fragment
 #[derive(Clone)]
 pub struct Uri {
-    source: Cow<'static, str>,
```

Comment:
Again, instead of using only `MemSlice`, I'd think an enum of static, owned, or shared would be the best here. This is because it's not free to convert to a `MemSlice` if we don't already have one. And we want to keep the optimization for a super common route, `/` being just a static constant.